daemon/logger/jsonfilelog/jsonfilelog_windows_test.go

```diff
@@ -0,0 +1,17 @@
+package jsonfilelog
+
+import (
+	"os"
+
+	"golang.org/x/sys/windows"
+)
+
+func isSharingViolation(err error) bool {
+	switch err := err.(type) {
+	case *os.SyscallError:
+		if err.Err == windows.ERROR_SHARING_VIOLATION {
```

Comment:
I _think_ this is the syscall error corresponding with the error we're running into, but perhaps someone working on a Windows machine can double-check

Reply:
By actual testing using a file locking utility [https://www.jensscheffler.de/filelocker], I observed that it is `os.PathError` and not `os.SyscallError`. 

Trying to print it: 
`Err3 &{%!!(string=open) %!!(string=ttr) %!!(syscall.Errno=32)}V`

So the check against `windows.ERROR_SHARING_VIOLATION` stays the same because its value is 32.

Why not bypass all this and just check the exact error string - just trying to learn best practices here.

Reply:
Errorstrings usually are for humans, and may change (or even be localised in the Windows case), so trying to avoid string matching where possible.

Perhaps this particular error is "stable", but we had a case (need to search for it) where the message changed in Windows RS4 or RS5, causing things to break